elytron/WFLY-14735-Management-FORM-authentication.adoc

```diff
@@ -0,0 +1,280 @@
+= WFLY-14735 FORM authentication for the management console with MFA
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The development of this RFE is delivering a subset of functionality from a much larger planned change,
+this RFE will start with two features:
+
+ * Enabling FORM authentication for admin console access to the application server.
+ * Where custom security realms are in use supporting MFA authentication on the login form to 
+ request additional attributes.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14735[WFLY-14735]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1696[EAP7-1696]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+The main development for this RFE will be a new `elytron-authentication` subsystem 
+in the https://github.com/wildfly/wildfly[WildFly] project.
+
+Changes will also be made within https://github.com/wildfly/wildfly-core[WildFly Core] to
+enable and support the activation of FORM authentication.
+
+Updates will be required to the https://github.com/hal/console[Admin Console] to support
+possibly being required to redirect to the login form and to make use of the logout endpoint.
+
+Security components developed for this RFE will be contained within the
+https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron] project.
+
+=== Other Interested Projects
+
+None
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+==== General Flow
+
+On the HTTP management interface a new context will be made available at the path `/authentication`,
```

Comment:
The context name could be the same as the subsystem name, for easier understanding/interlinking. The same is applied in case of `keycloak`/`elytron-oidc-client`, see https://github.com/wildfly/wildfly-proposals/pull/506/files.

Reply:
I almost agree but in this case the context is advertised by the management endpoint not the subsystem, i.e. the management endpoint will obtain a reference to a capability that can handle the HTTP requests but it is the management endpoint that choose where to register it.

Reply:
OK, then it makes sense to keep it as it is. In (now unexpected) case of adding next authentication related subsystem the endpoint would remain the same and just the underlying implementation would change.

Reply:
+1 being a capability reference we can swap the impl